### PR TITLE
Nano: Add openvino quantize support for Tensorflow

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Add openvino quantize support for Tensorflow

### 1. Why the change?

Support more features

### 2. User API changes

```python
from bigdl.nano.tf.keras import Model
model = Model(...)

# trace and quantize
openvino_model = model.trace(accelerator='openvino')
openvino_quantized_model = openvino_model.quantize(accelerator='openvino', calib_dataset=dataset)

# or quantize directly from tensorflow
openvino_quantized_model = model.quantize(accelerator='openvino', calib_dataset=train_dataset)
```

### 3. Summary of the change 

Add openvino quantize support for Tensorflow and an UT

### 4. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

**Comment by rnwang04** [comment]
LGTM

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py [hunk 1]
```diff
@@ -39,12 +39,14 @@ def quantize(self,
...
         Post-training quantization on a keras model.
 
-        :param calib_dataset:   A tf.data.Dataset object for calibration. Required for
-                                static quantization. It's also used as validation dataloader.
+        :param calib_dataset:   A tf.data.Dataset object for calibration without setting
+                                batch_size or batch_size=1. Required for static quantization.
+                                It's also used as validation dataloader.
         :param precision:       Global precision of quantized model,
                                 supported type: 'int8', 'bf16', 'fp16', defaults to 'int8'.
         :param accelerator:     Use accelerator 'None', 'onnxruntime', 'openvino', defaults to None.
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/tf/keras/inference_utils.py:52]: 'onnxruntime' could be removed since we have not support it.

**Review by TheaperDeng — APPROVED** [review]
only minor comment, we could merge this PR due to the bad github action condition and fix those things later

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py [hunk 2]
```diff
@@ -39,12 +39,14 @@ def quantize(self,
...
         """
         Post-training quantization on a keras model.
 
-        :param calib_dataset:   A tf.data.Dataset object for calibration. Required for
-                                static quantization. It's also used as validation dataloader.
+        :param calib_dataset:   A tf.data.Dataset object for calibration without setting
+                                batch_size or batch_size=1. Required for static quantization.
+                                It's also used as validation dataloader.
         :param precision:       Global precision of quantized model,
                                 supported type: 'int8', 'bf16', 'fp16', defaults to 'int8'.
```
[the remark this answers is not in the document]
Comment (jason-dai) [reply python/nano/src/bigdl/nano/tf/keras/inference_utils.py:51]: I think we will have bf16 support; why do we have fp16 here?

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py (hunk 1 above)
Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/tf/keras/inference_utils.py:52]: we support it in `trace` through `tf-onnx`, while not in `quantize` for now.

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/tf/keras/inference_utils.py:51]: yeah, fp16 won't happen in future either.

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py (hunk 1 above)
Comment (jason-dai) [reply python/nano/src/bigdl/nano/tf/keras/inference_utils.py:52]: Can we support tf->onnx->inc->ort?

File: python/nano/test/openvino/tf/test_openvino_quantize.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        assert y_hat.shape == (10, 10)
+
+        y_hat = openvino_quantized_model.predict(train_examples, batch_size=5)
+        assert y_hat.shape == (100, 10)
+
+        # Case2: Quantize directly from tensorflow
+        openvino_quantized_model = model.quantize(accelerator='openvino',
+                                                  calib_dataset=train_dataset)
+        
+        y_hat = openvino_quantized_model(train_examples[:10])
```
Comment (yangw1234) [inline python/nano/test/openvino/tf/test_openvino_quantize.py:46]: maybe also assert the dtype?

**Review by yangw1234 — APPROVED** [review]

File: python/nano/src/bigdl/nano/tf/keras/inference_utils.py (hunk 1 above)
Comment (MeouSker77) [reply python/nano/src/bigdl/nano/tf/keras/inference_utils.py:52]: We have supported tf onnxruntime quantize in another PR: #6461

**Comment by MeouSker77** [comment]
> only minor comment, we could merge this PR due to the bad github action condition and fix those things later

Ok, I'll merge this PR, and fix those things in another PR: #6461

**Pull request merged**